core/src/main/java/org/elasticsearch/ingest/PipelineStore.java

```diff
@@ -162,6 +161,25 @@ public ClusterState execute(ClusterState currentState) throws Exception {
         });
     }
 
+    void validatePipeline(Map<DiscoveryNode, IngestInfo> ingestInfos, PutPipelineRequest request) throws Exception {
```

Comment:
I agree on validation, I was just wondering if there was a more generic way to "check that a plugin is installed on all nodes". but this is a good one for ingest, to make sure that you store a pipeline only when it will work. That said you can store a pipeline, the stop some node, uninstall some plugin and start the node, but the pipeline is still there and not guaranteed to work?